packages/core/src/errors/core_error.rs

```diff
@@ -399,9 +402,9 @@ impl From<DivideByZeroError> for StdError {
 ///
 /// The prefix "Std" means "the standard result within the standard library". This is not the only
 /// result/error type in cosmwasm-std.
-pub type StdResult<T> = core::result::Result<T, StdError>;
+pub type CoreResult<T, E = CoreError> = core::result::Result<T, E>;
```

Comment:
> Why the new generic?

Oh, it's just personal preference when I write `Result` aliases, no real reason behind it.  
I can revert it if needed.

Rationale here is, usually I call result aliases I write `Result` itself, and with that kind of definition both of these are valid

- `Result<T>`
- `Result<T, SomeError>`

Making it more versatile overall

Reply:
> Also the doc comment might need some rewording, since it does not have the Std prefix in cosmwasm-core, only in cosmwasm-std.

Yeah, might need some rewording, I'll probably go for the middle route here, changing the wording from

> `Std` prefix

to

> `Core`/`Std` prefix, since it's a straight re-export from core, and we don't want to potentially confuse people.

Reply:
Yeah, makes sense to me

Reply:
Regarding the Result: I see now where you are coming from. With an alias named `Result` it makes a lot of sense, as you can avoid having to come up with a new name (like `StdResult` 😉).
I think since we are already stuck with `StdResult`, it's less helpful, since you can still use the normal `Result` without any import shenanigans, even when the alias is imported.
The name `StdResult` kind of implies `Result` with an `StdError`, so I would slightly prefer the hardcoded one in this case.

Reply:
Changed the documentation and reverted the change with the default generic type.